src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -491,6 +491,10 @@ private IEnumerable<TypeInferenceInfo> InferTypeInObjectCreationExpression(BaseO
                 }
 
                 return InferTypeInArgument(index, methods, argumentOpt, invocation);
+
+                static bool MethodAcceptsEmptyArgument(IMethodSymbol methodSymbol)
+                    => methodSymbol.Parameters.Length == 0 ||
+                       methodSymbol.Parameters.Length == 1 && methodSymbol.Parameters[0].IsParams;
```

Comment:
please break into different statements, or parenthesize for clarity.

Reply:
what about a method where all the parameter are optional?  woudl that be a case we need to care about?  even if not, can you add a test where one of the overloads has an optional parameter?

thanks!

Reply:
Good catch! Fixed